scripts/nix/default.nix

```diff
@@ -0,0 +1,134 @@
+{
+  stdenv,
+  lib,
+  rustPlatform,
+  # Allows overriding for the plugins
+  pname ? "nushell",
+  # Use this for the cargo file that the plugin has
+  extraCargo ? ../../Cargo.toml,
+  withDefaultFeatures ? true,
+  additionalFeatures ? (p: p),
+  # Package dependencies
+  pkg-config,
+  openssl,
+  curlMinimal,
+  python3,
+  xorg,
+  nghttp2,
+  libgit2,
+  zstd,
+  zlib,
+  nativeBuildInputs ? [
+    pkg-config
+  ]
+  ++ lib.optionals (withDefaultFeatures && stdenv.hostPlatform.isLinux) [ python3 ]
+  ++ lib.optionals stdenv.cc.isClang [ rustPlatform.bindgenHook ],
+  buildInputs ? [
+    zstd
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [ zlib ]
+  ++ lib.optionals (withDefaultFeatures && stdenv.hostPlatform.isLinux) [ xorg.libX11 ]
+  ++ lib.optionals (withDefaultFeatures && stdenv.hostPlatform.isDarwin) [
+    nghttp2
+    libgit2
+  ],
+  checkInputs ?
+    lib.optionals stdenv.hostPlatform.isDarwin [ curlMinimal ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [ openssl ],
+
+  # The skipped tests all fail in the sandbox because in the nushell test playground,
+  # the tmp $HOME is not set, so nu falls back to looking up the passwd dir of the build
+  # user (/var/empty). The assertions however do respect the set $HOME.
+  testsToSkip ? [
+    "repl::test_config_path::test_default_config_path"
+    "repl::test_config_path::test_xdg_config_bad"
+    "repl::test_config_path::test_xdg_config_empty"
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    "plugins::config::some"
+    "plugins::stress_internals::test_exit_early_local_socket"
+    "plugins::stress_internals::test_failing_local_socket_fallback"
+    "plugins::stress_internals::test_local_socket"
```

Comment:
Could you open an issue for fixing these tests and include details about how they fail? I'd do it myself, but I don't have a Mac. Ideally, we shouldn't be ignoring tests. This isn't just for Nix - contributors with weirder setups should still be able to develop Nushell easily.

I've already put up a PR (https://github.com/nushell/nushell/pull/16914) to address the other three failing tests (the `test_config_path` ones), so you should be able to remove that bit once it's merged.

Reply:
I didn't find this, but I'll look into why they're failing. Usually it's a sandbox problem for nix stuff, there's just some stuff that doesn't work well in there.